src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -517,7 +517,7 @@ private static void ApplyLibContract(LocalPackageInfo package, LockFileTargetLib
                 lockFileLib.Dependencies = dependencies
                     .Where(ld => ld.LibraryRange.TypeConstraintAllowsAnyOf(LibraryDependencyTarget.PackageProjectExternal))
                     .Select(ld => new PackageDependency(ld.Name, ld.LibraryRange.VersionRange))
-                    .ToList();
+                    .AsList();
```

Comment:
This change is unnecessary. There's no functional change here. 
LINQ generates enumerates which had to be converted to List.